src/App.js

```diff
@@ -0,0 +1,36 @@
+import DocumentPage from "./pages/DocumentPage.js";
+import EditorPage from "./pages/EditorPage.js";
+import { initRouter } from "./utils/router.js";
+
+export default function App({$target}){
+    const documentPage = new DocumentPage({
+        $target,
+    })
+
+    const editorpage = new EditorPage({
+        $target,
+        initialState :{}
+    })
+
+    this.route = () =>{
+        const {pathname} = window.location
+        $target.innerHTML = ""
+        if(pathname === "/"){
+            documentPage.render()
+        }else if(pathname.indexOf("/documents/") === 0){
+            const [,,documentId] = pathname.split("/")
+            documentPage.render()
+            editorpage.setState({documentId})
+        }else{
+            $target.innerHTML = ""
+            documentPage.render()
+        }
```

Comment:
```suggestion
        $target.innerHTML = ""
        documentPage.render()
    
        if(pathname.indexOf("/documents/") === 0){
            const [,,documentId] = pathname.split("/")
            editorpage.setState({documentId})
        }
```

공통된 부분이 많아 이렇게 줄여볼 수 있겠네요~